static_frame/core/index.py

```diff
@@ -430,6 +430,12 @@ def __init__(self,
         # NOTE: automatic discovery is possible, but not yet implemented
         self._loc_is_iloc = loc_is_iloc
 
+    def __reversed__(self) -> tp.Iterator[tp.Hashable]:
+        '''
+        Returns a reverse iterator on the index labels.
+        '''
+        return reversed(self.values)
```

Comment:
On second thought, it's probably better to bypass the descriptor altogether and just do what `__iter__` is doing:

```suggestion
        if self._recache:
            self._update_array_cache()
        return reversed(self._labels)
```